server/pulp/server/webservices/views/search.py

```diff
@@ -0,0 +1,121 @@
+    1) If it's for an "old-style" model, manager must be defined.
+
+    :cvar    generate_json_response: The function that should be used to turn the search results
+                                     into a JSON serialized Django Response object. If not defined,
+                                     this defaults to
+                                     pulp.server.webservices.views.util.generate_json_response.
+    :vartype generate_json_response: staticmethod
+    :cvar    manager:                Define this class attribute if you are making a SearchView for
+                                     a model that has not yet been converted to MongoEngine. It
+                                     should represent an instance of the Model's manager class,
+                                     which must have a find_by_criteria() method.
+    :vartype manager:                object
+    :cvar    model:                  Define this class attribute if you are making a SearchView for
+                                     a MongoEngine Document. The model must have a meta class
+                                     attribute defined with the 'queryset_class' key indexing
+                                     pulp.server.db.model.base.CriteriaQuerySet, so that the
+                                     Document's QuerySet will have a find_by_criteria() method.
+    :vartype model:                  mongoengine.Document
+    :cvar    serializer:             If your view needs to modify the QuerySet results before they
+                                     are returned to the caller, you can define this attribute. It
+                                     should be set to a function that accepts a single object, and
+                                     returns a single object.
+    :vartype serializer:             staticmethod
+    """
+    generate_json_response = staticmethod(util.generate_json_response)
+
+    @auth_required(authorization.READ)
+    def get(self, request):
+        """
+        Search for objects using an HTTP GET request.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :return:        HttpReponse containing a list of objects that were matched by the request
+        :rtype:         django.http.HttpResponse
+        """
+        query = dict(request.GET)
+
+        fields = query.pop('field', '')
+        if fields:
+            query['fields'] = fields
+
+        return self._generate_response(query)
+
+    @auth_required(authorization.READ)
+    def post(self, request):
+        """
+        Search for objects using an HTTP POST request.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :return:        HttpReponse containing a list of objects that were matched by the request
+        :rtype:         django.http.HttpResponse
+        """
+        query = json.loads(request.body)
```

Comment:
You can use the `@json_body_required` decorator instead. This will enforce the existence of a body, validate the json, populate a field `request.body_as_json`.